### ISSUE TITLE
The results I obtained are far from the performance mentioned in your paper

### ISSUE THREAD

**Issue by Tingyangll** [SamsungLabs/unified-demosaicing#2]
Hello, I am very interested in your work. I would like to ask a question: After obtaining your dataset and downloading the "RCAN" model, I used your weights for inference, but the results I obtained are far from the performance mentioned in your paper. Could you advise on the direction I should take to resolve this issue? Your response is crucial to me. Thank you for your assistance. (Note: My results on the Hard patches dataset are as follows: ->Single_Mean_MSE: 0.0006, ->Single_Mean_PSNR: 32.06, ->Single_Mean_SSIM: 0.83, ->Single_Mean_Delta: 8.74. Additionally, my results on the Full images dataset are as follows: ->Single_Mean_MSE: 0.0003, ->Single_Mean_PSNR: 34.41, ->Single_Mean_SSIM: 0.81, ->Single_Mean_Delta: 11.74.)

**Comment by tedlasai** [comment]
Are you utilizing our evaluation code? Or your own evaluation script?

**Comment by tedlasai** [comment]
This looks quite strange... Are you getting this for all the checkpoints? Or just this one? Also I wonder if it has to do with pytorch versions.. Are you using our enviroment.yaml file?

**Comment by tedlasai** [comment]
I uploaded the git with rcan/common file for you to try. I am trying to replicate the issue now... The only other thing is perhaps the data I uploaded has some issue.. Lets see

**Comment by tedlasai** [comment]
Great to hear its resolved. For the drop 0.01, I didn't provide checkpoints to models w/ our maskout augmentation. I apologize. If you are unable to train it yourself let me know, and I might be able to provide new checkpoints for you in some time (but probably not for a few months, I have got some deadlines). But haven't retrained in our refactored code base for the public release.

Next. GT images are in RAW and the dimensions are 48x48x3 -> We captured in pixelshift, so we have GT RawRGB at every pixel. we averaged the two green measurements as pixelshift actually gives you 4 at each piuxel.

We are demosaicing so the ground truth should be 3 channel. The output is between 0 and 13496 (white_level-black_level). However, for training we divide by 13496 so the inputs are normalized between 0 and 1.. You can check the dataloader for more details.

But this model outputs RAW data and is trained only RAW data, that is why we need the dng to produce renderings in our visualization code.

**Comment by tedlasai** [comment]
I have started retraining so I can include checkpoints w/ maskout augmentation!

**Comment by tedlasai** [comment]
Give me a few days for models to train!

**Comment by tedlasai** [comment]
Ok great, so looks like everything works from the code! Wonderful! 

 RAW denoising or demosaicing models are very specific to the type of noise and the sensor itself. Thus, I recommend fine-tuning on those datasets. Generalized models do exist, but typically don't work as well if the noise or the sensor changes.

Also, are you testing on mosaics from SIDD? Remember, our model takes in a 4-channel input, one-channel mosaic and three-channels of pattern embeddings.

Last, I am closing this issue as it is solved. Feel free to continue discussion with me in either a new github issue or by email.
